### PR TITLE
Logic Integration: ClientService ModifyWorkout

### LINKED ISSUE #99: Logic Integration: ClientService - Core methods implementation
﻿### Task
Integrate logic removed from Client.cs.

### Requirements
* Implementation: Implement the logic for SetWorkout() and ModifyWorkout() within the ClientService layer.

### PR DESCRIPTION
**Pull request by AndaChiorean**
## Summary
- Added ModifyWorkoutAsync to ClientService for updating existing workout logs through the repository layer
- FinalizeWorkoutAsync already handles creating new workout logs (covers the original SetWorkout stub), so ModifyWorkout completes the pair
- Added PUT endpoint at api/client/modify-workout for the new operation
- Added 6 unit tests covering valid modifications, null/invalid input handling, missing client scenarios, and FinalizeWorkout edge cases

Closes #99

### CODE REVIEW

File: UBB-SE-2026-832-1.sln [hunk 1]
```diff
@@ -1,34 +1,76 @@
+﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 17
 VisualStudioVersion = 17.0.31903.59
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ClassLibrary", "ClassLibrary\ClassLibrary.csproj", "{1239CA30-0B9C-4962-B6A2-C8C7A9700B2B}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebAPI", "WebAPI\WebApi.csproj", "{FD9CC266-52C8-4F13-BC6F-C2BC714E6F67}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebApi", "WebAPI\WebApi.csproj", "{FD9CC266-52C8-4F13-BC6F-C2BC714E6F67}"
```
Comment (MarioMaxim18) [inline UBB-SE-2026-832-1.sln]: Same here, keep WebAPI

**Review by MarioMaxim18 — APPROVED** [review]

File: Tests/ClientServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    [Fact]
+    public async Task ModifyWorkout_WhenClientNotFound_ReturnsFalse()
+    {
+        this.clientRepo.Setup(r => r.GetByIdAsync(999)).ReturnsAsync((Client?)null);
+
+        var dto = new WorkoutLogDataTransferObject
+        {
+            WorkoutLogId = 10,
+            Client = new ClientDataTransferObject { ClientId = 999 },
+            WorkoutName = "Ghost Workout",
```
Comment (MarioMaxim18) [inline Tests/ClientServiceTests.cs:90]: we should have const for tests

**Review by MarioMaxim18 — CHANGES_REQUESTED** [review]

File: UBB-SE-2026-832-1.sln (hunk 1 above)
Comment (AndaChiorean) [reply UBB-SE-2026-832-1.sln]: Fixed, same dotnet sln issue. Restored to WebAPI.

File: Tests/ClientServiceTests.cs (hunk 2 above)
Comment (AndaChiorean) [reply Tests/ClientServiceTests.cs:90]: The convention says to use const on variables that could be const -- it does not say to extract every inline string literal into a named constant. These are scenario-specific test values that describe the test case, not reused magic strings. Extracting them to constants at the top of the file forces readers to scroll up to understand each test, which hurts readability. Test data should be visible inline so the method reads as documentation.

**Review by MarioMaxim18 — APPROVED** [review]
